controllers/argocd/deployment.go

```diff
@@ -350,6 +343,59 @@ func isMergable(extraArgs []string, cmd []string) error {
 	return nil
 }
 
+// appendUniqueArgs appends extraArgs to cmd while ignoring any duplicate flags.
```

Comment:
I see this PR has similar unit tests for other functions, but can we also add a unit test for this function? 

The logic in this function is complex (given the large possible variation in inputs), so I'd like to ensure we have an exhaustive list of test cases that verify the possible cases we can think of that this function might be asked to handle.